extension/content/contexts/environment.js

```diff
@@ -276,10 +336,16 @@ export function useSelectedEnvironmentAuth() {
   return state.auth[state.selectedKey];
 }
 
+export function useSelectedEnvironmentConnectionStatus() {
+  const { state } = React.useContext(environmentContext);
+  return state.connectionStatus[state.selectedKey];
+}
+
 export function useSelectedNormandyEnvironmentAPI() {
   const environment = useSelectedEnvironment();
   const auth = useSelectedEnvironmentAuth();
-  return new NormandyAPI(environment, auth);
+  const connectionStatus = useSelectedEnvironmentConnectionStatus();
```

Comment:
Why are these three separate things? Why not something like

```js
const {environment, auth, connectionStatus} = useSelectedEnvironmentInfo()
```

As the context grows it seems strange to keep adding these very specific selectors. Is there some advantage to it?

Reply:
yeah i think i had already done it so i was repeating the pattern but i was also starting to feel like there were too many of these